src/Microsoft.TemplateEngine.Authoring.TemplateVerifier/VerificationEngine.cs

```diff
@@ -0,0 +1,335 @@
+        { }
+
+        private static async IAsyncEnumerable<(string FilePath, string ScrubbedContent)> GetVerificationContent(string contentDir, List<Glob> globs, ScrubbersDefinition? scrubbers)
+        {
+            foreach (string filePath in Directory.EnumerateFiles(contentDir, "*", SearchOption.AllDirectories))
+            {
+                if (globs.Any(g => g.IsMatch(filePath)))
+                {
+                    continue;
+                }
+
+                string content = await File.ReadAllTextAsync(filePath).ConfigureAwait(false);
+
+                if (scrubbers != null)
+                {
+                    string extension = Path.GetExtension(filePath);
+
+                    if (string.IsNullOrEmpty(extension) || !scrubbers.ScrubersByExtension.TryGetValue(extension, out Action<StringBuilder>? scrubber))
+                    {
+                        scrubber = scrubbers.GeneralScrubber;
+                    }
+
+                    if (scrubber != null)
+                    {
+                        var sb = new StringBuilder(content);
+                        scrubber(sb);
+                        content = sb.ToString();
+                    }
+                }
+
+                yield return new(filePath, content);
+            }
+        }
+
+        private static Task CreateVerificationTask(string contentDir, TemplateVerifierOptions options)
+        {
+            List<string> exclusionsList = (options.DisableDefaultVerificationExcludePatterns ?? false)
+                ? new()
+                : new(_defaultVerificationExcludePatterns);
+
+            if (options.VerificationExcludePatterns != null)
+            {
+                exclusionsList.AddRange(options.VerificationExcludePatterns);
+            }
+
+            List<Glob> globs = exclusionsList.Select(pattern => Glob.Parse(pattern)).ToList();
+
+            SettingsTask defaultVerifyTask = Verifier.VerifyDirectory(
+                contentDir,
+                (filePath) => !globs.Any(g => g.IsMatch(filePath)));
+
+            if (options.CustomScrubbers != null)
+            {
+                if (options.CustomScrubbers.GeneralScrubber != null)
+                {
+                    defaultVerifyTask = defaultVerifyTask.AddScrubber(options.CustomScrubbers.GeneralScrubber);
+                }
+
+                foreach (var pair in options.CustomScrubbers.ScrubersByExtension)
+                {
+                    defaultVerifyTask = defaultVerifyTask.AddScrubber(pair.Key, pair.Value);
+                }
+            }
+
+            if (options.CustomDirectoryVerifier != null)
+            {
+                return options.CustomDirectoryVerifier(
+                    contentDir,
+                    new Lazy<IAsyncEnumerable<(string FilePath, string ScrubbedContent)>>(
+                        GetVerificationContent(contentDir, globs, options.CustomScrubbers)));
+            }
+
+            Verifier.DerivePathInfo(
```

Comment:
this api is designed to be called once a startup. usually in a `[ModuleInitializer]`